test/test_trio_server/mock_lean.py

```diff
@@ -0,0 +1,152 @@
+"""
+This makes it possible to have tests for the Lean Python server
+interface which do not depend on having Lean installed in a certain manner.
+
+Instead one can build a "lean script" which is run by a fake lean server.
+"""
+
+import trio  # type: ignore
+from typing import List, Dict
+import json
+from collections import deque
+from dataclasses import dataclass
+import trio.testing
+
+from lean_client.trio_server import TrioLeanServer
+
+
+class LeanScriptStep:
+    pass
+
+
+@dataclass
+class LeanTakesTime(LeanScriptStep):
+    """
+    Simulate taking a while to start up or processes a request.
+    """
+    seconds: float
```

Comment:
If you use `@dataclass` with a single member and never subclass afterwards then I think you'd be better off using [NewType](https://mypy.readthedocs.io/en/stable/more_types.html#newtypes) (but below I'll suggest to add a method here, so this is a virtual comment).

Reply:
I'm not sure I see the advantage here of NewType in terms of performance (this code doesn't need to be performant) or readability, but sure I can switch them (assuming we don't go with the `.run()` method suggestion from above.)

Reply:
Is it possible to use NewType here since it is a subclass of LeanScriptStep?

Reply:
In that scenario it wouldn't be a subclass.